x-pack/plugins/upgrade_assistant/public/application/components/overview/identify_step/use_deprecation_logging.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { useAppContext } from '../../../app_context';
+import { ResponseError } from '../../../lib/api';
+import { DeprecationLoggingPreviewProps } from '../../types';
+
+const i18nTexts = {
+  enabledMessage: i18n.translate(
+    'xpack.upgradeAssistant.overview.deprecationLogs.enabledToastMessage',
+    {
+      defaultMessage: 'Log deprecated actions.',
+    }
+  ),
+  disabledMessage: i18n.translate(
+    'xpack.upgradeAssistant.overview.deprecationLogs.disabledToastMessage',
+    {
+      defaultMessage: 'Do not log deprecated actions.',
+    }
+  ),
+};
+
+export const useDeprecationLogging = (): DeprecationLoggingPreviewProps => {
+  const { api, notifications } = useAppContext();
+  const { data, error: fetchError, isLoading, resendRequest } = api.useLoadDeprecationLogging();
+
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [hasLoggerDeprecationWarning, setLoggerDeprecationWarning] = useState(false);
```

Comment:
This component doesn't know how it's being consumed, so in the current context "warning" doesn't carry much meaning. I think it's clearer to reason about this as "logs are stored in the index", yielding a name like:

```js
const [isDeprecationLogIndexingEnabled, setIsDeprecationLogIndexingEnabled] = useState(undefined);
```

This also continues the naming suggestion I made server-side.